docs/DeveloperGuide.md

```diff
@@ -227,6 +225,46 @@ From the notes in the sequence diagram above, for every attribute in the instrum
 prompt to get user to provide information for that attribute. This is done through a series of methods in
 the `TextUi` class.
 
+### Loading pre-existing data
+The loading of pre-existing data is mainly handled by the `filemanager` and `model` components. The main method calls 
+`Storage#loadFileData(instrumentManager)` which uses `InstrumentDecoder#readFile(instrumentManager, fileData)`. This method calls 
+`InstrumentDecoder#addSavedInstrumentToList(instrumentManager, textSegment)` for each pre-existing instrument which will add the 
+corresponding instrument in the `InstrumentManager` through calling the `XYZDecoder#addXYZToList`. In the event
+the instrument is not one of the 4 types of instruments, the `InstrumentDecoder` will throw a new `InvalidInstrumentInFileError`
+and display the corresponding error message.
+
+The figures below represents the sequence diagrams when the user loads a pre-existing crypto:
+
+<img src="images/FileManagerSeqBetweenStorageAndDecoder.png" width="800"/>
+
+More details about the reference frame for decoding and updating the `InstrumentManager` is shown below:
+
+<img src="images/FileManagerSequenceDiagram.png" width="1080" height="500"/>
+
+More details about the reference frame for adding the decoded instrument into the `InstrumentManager` is shown below:
+
+<img src="images/FileManagerRefDiagCryptoDecoder.png" width="950"/>
+
+The process for loading other pre-existing instruments follow a similar process to the sequence above. The main difference
+would be the type of instrument decoder called, the different instrument specific decoded parameters and the type of instrument
+added to the `InstrumentManager`. For example when loading a stock instead of calling `CryptoDecoder#addCryptoToList(textSegment, instrumentManager`
```

Comment:
```suggestion
added to the `InstrumentManager`. For example when loading a stock instead of calling `CryptoDecoder#addCryptoToList(textSegment, instrumentManager)`
```